### PR TITLE
ENT-2017 Document for multiple enterprises decision

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
JIRA: https://openedx.atlassian.net/browse/ENT-2016
          https://openedx.atlassian.net/browse/ENT-2017

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

File: docs/decisions/0001-multiple-enterprise-support.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#. When a learner is linked with an enterprise and tries to link with another enterprise(through SSO), the learner is redirected to the login page instead of the registration page for the second enterprise. This is because the system detects that these credentials are already in the system.
+#. Enrollment, Progression, Completion, Consent records become ambiguous if a learner completes a course from enterprise A and then attempts the same course through enterprise B.
+#. Incorrect discounts are applied when an employee linked with multiple enterprises redeems the benefit applicable only to one of the enterprises.
+
+
+Decisions
+=========
+
+Short term:
+#. As part of the login flow the user would select one of the linked enterprises.
```
Comment (muhammad-ammar) [inline docs/decisions/0001-multiple-enterprise-support.rst]: Can we some information regarding `What steps a logged in learner has to do to switch to another enterprise?`

Comment (irfanuddinahmad) [reply docs/decisions/0001-multiple-enterprise-support.rst]: Some information is added in the "Long term" section. More detail would be added as part of the detailed solution.

**Comment by irfanuddinahmad** [comment]
> It would be great if we can add information regarding
> 
> 1. What will happen in the case of API or server-to-server calls enterprise, How we will decide what is the current enterprise context?
> 2. What to do in a case when we need context independent response? For example, Return all the enterprises of a learner.
> 3. How the current enterprise context is stored? Cache/Session/DB?

The current enterprise context could be part of Session, JWT etc. The answers to the above queries would be dependent on the detailed solution.

**Comment by brittneyexline** [comment]
is there an ETA on when this is getting merged? i ask because there is another ADR that is in progress. thanks!

**Comment by irfanuddinahmad** [comment]
> is there an ETA on when this is getting merged? i ask because there is another ADR that is in progress. thanks!

@brittneyexline I think this can be merged if I receive another approval. Can I request you for a review?

**Comment by irfanuddinahmad** [comment]
> i don't have any particular feedback about the content of this, however when i viewed this in rich text, the formatting appeared to be messed up. i would recommend double checking that and fixing appropriately before merging.

@brittneyexline The formatting seems to be fine in gedit. Is rich text a requirement for ADRs?

**Pull request merged**